fdbserver/DataDistribution.actor.cpp

```diff
@@ -755,68 +755,45 @@ struct DDTeamCollection : ReferenceCounted<DDTeamCollection> {
 
 			int64_t bestLoadBytes = 0;
 			Optional<Reference<IDataDistributionTeam>> bestOption;
-			std::vector<std::pair<int, Reference<IDataDistributionTeam>>> randomTeams;
-			std::set< UID > sources;
-
+			std::vector<Reference<IDataDistributionTeam>> randomTeams;
+			std::set<UID> completeSources;
+			for( int i = 0; i < req.completeSources.size(); i++ ) {
+				completeSources.insert( req.completeSources[i] );
+			}
```

Comment:
And also if you use this constructor, you should be able to mark it as `const`, which could clear up any ambiguity about whether you are duplicating the original data so that you can modify it or to just use as a lookup.